src/m365/spo/commands/site/site-admin-add.spec.ts

```diff
@@ -0,0 +1,693 @@
+      ) {
+        return;
+      }
+
+      throw 'Invalid request: ' + opts.url;
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl, userId: adminToAddId, primary: true, verbose: true } });
+    assert(loggerLogToStderrSpy.firstCall.firstArg === 'Adding site administrator...');
+  });
+
+  it('adds a group as primary site collection admins by userName as admin with verbose parameter', async () => {
+    let result = false;
+    let primaryAdminResult = false;
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/root?$select=webUrl`) {
+        return { res: { webUrl: rootUrl } };
+      }
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups?$filter=displayName eq '${groupName}'`) {
+        return {
+          value: [{
+            mail: undefined,
+            id: groupId
+          }]
+        };
+      }
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/contoso.sharepoint.com:/sites/site?$select=id`) {
+        return { id: `contoso.sharepoint.com,${siteId},fb0a066f-c10f-4734-94d1-f896de4aa484` };
+      }
+
+      throw 'Invalid request: ' + opts.url;
+    });
+
+    sinon.stub(request, 'post').callsFake(async opts => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/GetSiteAdministrators?siteId='${siteId}'`) {
+        return JSON.stringify({
+          value: listOfAdminsFromAdminSource
+        });
+      }
+
+      if (opts.url === `${adminUrl}/_api/SPOInternalUseOnly.Tenant/SetSiteSecondaryAdministrators` && opts.data.secondaryAdministratorsFieldsData) {
+        result = opts.data.secondaryAdministratorsFieldsData.siteId === siteId &&
+          opts.data.secondaryAdministratorsFieldsData.secondaryAdministratorLoginNames.length === 3 &&
+          opts.data.secondaryAdministratorsFieldsData.secondaryAdministratorLoginNames[0] === 'i:0#.f|membership|user1loginName@email.com' &&
+          opts.data.secondaryAdministratorsFieldsData.secondaryAdministratorLoginNames[1] === 'i:0#.f|membership|user2loginName@email.com' &&
+          opts.data.secondaryAdministratorsFieldsData.secondaryAdministratorLoginNames[2] === `c:0t.c|tenant|${groupId}`;
+        return;
+      }
+
+      throw 'Invalid request: ' + opts.url;
+    });
+
+    sinon.stub(request, 'patch').callsFake(async opts => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/sites('${siteId}')`) {
+        primaryAdminResult = opts.data.Owner === `c:0t.c|tenant|${groupId}` && opts.data.SetOwnerWithoutUpdatingSecondaryAdmin === true;
+        return;
+      }
+
+      throw `Invalid PATCH request: ${JSON.stringify(opts, null, 2)}`;
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl, groupName: groupName, asAdmin: true, primary: true, verbose: true } });
+    assert(result);
+    assert(primaryAdminResult);
+    assert(loggerLogToStderrSpy.firstCall.firstArg === 'Adding site administrator as an administrator...');
+  });
+});
```

Comment:
Same as above here, but let's also use patchStub to check that as well.

Reply:
done